Automattic/woocommerce-payments
# Express checkout button display class

#### Changes proposed in this Pull Request

The goal of this PR is to combine the logic of the express checkout button display and the separator that follows the button. The logic is currently duplicated across multiple files and is prone to errors that could lead to undesired display issues. This centralizes the logic into a new class, `WC_Payments_Express_Checkout_Button_Display_Handler`. This class will handle initialization and logic for any express payment method buttons. 

This class doesn't add a new class to help prevent redundancy when adding new express payment methods, as mentioned in 1545-gh-Automattic/woopay. 

Closes #5472 









#### Testing instructions

The goal of testing should be to ensure that WooPay buttons and Payment Request Buttons all continue to function as expected and that the `-or-` separator is never duplicated or displaying when it shouldn't be.

1. Enable WooPay buttons and Payment Requests
2. Check that buttons are appearing in the appropriate locations(Product page, Cart page, Checkout page).
3. In the WCPay settings, toggle on/off button display on different pages and ensure those settings are respected.
4. Test that payments via each button are working as expected. This guide works well for testing Apple Pay: paJDYF-1Ys-p2 . For WooPay, a successful redirect to WooPay is enough.




-------------------


**Post merge**



- [] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
